Unable to upgrade to Spring Boot 2.4.6

As can be seen in https://github.com/vaadin/spring/pull/809

Moving from Spring Boot 2.4.5 to 2.4.6 causes the build to fail with
```
[Step 2/6] [INFO] --- maven-jar-plugin:3.0.2:jar (default-jar) @ vaadin-test-spring-boot-contextpath ---
[15:59:56][Step 2/6] [INFO] Building jar: /opt/agent/work/cd0aca6139b3f1ae/vaadin-spring-tests/test-spring-boot-contextpath/target/vaadin-test-spring-boot-contextpath-18.0-SNAPSHOT.jar
[15:59:56][Step 2/6] [INFO] 
[15:59:56][Step 2/6] [INFO] --- spring-boot-maven-plugin:2.4.6:start (pre-integration-test) @ vaadin-test-spring-boot-contextpath ---
[15:59:56][Step 2/6] [INFO] Attaching agents: []
[15:59:56][Step 2/6] Error: Exception thrown by the agent : java.rmi.server.ExportException: Port already in use: 9001; nested exception is: 
[15:59:56][Step 2/6] 	java.net.BindException: Address already in use (Bind failed)
[15:59:56][Step 2/6] sun.management.AgentConfigurationError: java.rmi.server.ExportException: Port already in use: 9001; nested exception is: 
[15:59:56][Step 2/6] 	java.net.BindException: Address already in use (Bind failed)
[15:59:56][Step 2/6] 	at sun.management.jmxremote.ConnectorBootstrap.startRemoteConnectorServer(ConnectorBootstrap.java:480)
[15:59:56][Step 2/6] 	at sun.management.Agent.startAgent(Agent.java:262)
[15:59:56][Step 2/6] 	at sun.management.Agent.startAgent(Agent.java:452)
[15:59:56][Step 2/6] Caused by: java.rmi.server.ExportException: Port already in use: 9001; nested exception is: 
[15:59:56][Step 2/6] 	java.net.BindException: Address already in use (Bind failed)
[15:59:56][Step 2/6] 	at sun.rmi.transport.tcp.TCPTransport.listen(TCPTransport.java:346)
[15:59:56][Step 2/6] 	at sun.rmi.transport.tcp.TCPTransport.exportObject(TCPTransport.java:254)
[15:59:56][Step 2/6] 	at sun.rmi.transport.tcp.TCPEndpoint.exportObject(TCPEndpoint.java:411)
[15:59:56][Step 2/6] 	at sun.rmi.transport.LiveRef.exportObject(LiveRef.java:147)
[15:59:56][Step 2/6] 	at sun.rmi.server.UnicastServerRef.exportObject(UnicastServerRef.java:236)
[15:59:56][Step 2/6] 	at sun.rmi.registry.RegistryImpl.setup(RegistryImpl.java:213)
[15:59:56][Step 2/6] 	at sun.rmi.registry.RegistryImpl.<init>(RegistryImpl.java:173)
[15:59:56][Step 2/6] 	at sun.management.jmxremote.SingleEntryRegistry.<init>(SingleEntryRegistry.java:49)
[15:59:56][Step 2/6] 	at sun.management.jmxremote.ConnectorBootstrap.exportMBeanServer(ConnectorBootstrap.java:816)
[15:59:56][Step 2/6] 	at sun.management.jmxremote.ConnectorBootstrap.startRemoteConnectorServer(ConnectorBootstrap.java:468)
[15:59:56][Step 2/6] 	... 2 more
[15:59:56][Step 2/6] Caused by: java.net.BindException: Address already in use (Bind failed)
[15:59:56][Step 2/6] 	at java.net.PlainSocketImpl.socketBind(Native Method)
[15:59:56][Step 2/6] 	at java.net.AbstractPlainSocketImpl.bind(AbstractPlainSocketImpl.java:387)
[15:59:56][Step 2/6] 	at java.net.ServerSocket.bind(ServerSocket.java:375)
[15:59:56][Step 2/6] 	at java.net.ServerSocket.<init>(ServerSocket.java:237)
[15:59:56][Step 2/6] 	at java.net.ServerSocket.<init>(ServerSocket.java:128)
[15:59:56][Step 2/6] 	at sun.rmi.transport.proxy.RMIDirectSocketFactory.createServerSocket(RMIDirectSocketFactory.java:45)
[15:59:56][Step 2/6] 	at sun.rmi.transport.proxy.RMIMasterSocketFactory.createServerSocket(RMIMasterSocketFactory.java:345)
[15:59:56][Step 2/6] 	at sun.rmi.transport.tcp.TCPEndpoint.newServerSocket(TCPEndpoint.java:666)
[15:59:56][Step 2/6] 	at sun.rmi.transport.tcp.TCPTransport.listen(TCPTransport.java:335)
[15:59:56][Step 2/6] 	... 11 more
[16:00:17][Step 2/6] Failed to execute goal org.springframework.boot:spring-boot-maven-plugin:2.4.6:start (pre-integration-test) on project vaadin-test-spring-boot-contextpath: Could not figure out if the application has started
```


Before the failure, the following modules pass
```
[Step 2/6] [INFO] vaadin-spring-parent 18.0-SNAPSHOT ................. SUCCESS [  0.491 s]
[INFO] vaadin-spring ...................................... SUCCESS [  9.947 s]
[INFO] Vaadin Spring tests ................................ SUCCESS [  6.999 s]
[INFO] Flow Spring Common UI components ................... SUCCESS [  3.247 s]
[INFO] Vaadin Spring Boot integration tests ............... SUCCESS [ 25.226 s]
[INFO] Vaadin Spring Boot integration tests with custom packages to scan SUCCESS [ 20.269 s]
[INFO] Vaadin Spring Boot deployable integration tests .... SUCCESS [ 22.882 s]
```

Presumably one of them leaves something running.

This can be reproduce locally also by running e.g.
```
mvn install -DskipTests
```
in `vaadin-spring-tests`

However, after Maven has terminated, nothing is left running

--- a/pom.xml
+++ b/pom.xml
@@ -44,7 +44,7 @@
 
         <!-- These are typically overridden with BOMs -->
         <vaadin.flow.version>8.0-SNAPSHOT</vaadin.flow.version>
-        <spring-boot.version>2.4.5</spring-boot.version>
+        <spring-boot.version>2.5.1</spring-boot.version>
 
         <!-- Additional manifest fields -->
         <Vaadin-License-Title>Apache License 2.0</Vaadin-License-Title>

--- a/vaadin-spring-tests/test-spring-common/pom.xml
+++ b/vaadin-spring-tests/test-spring-common/pom.xml
@@ -85,6 +85,11 @@
             <optional>true</optional>
         </dependency>
 
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-jetty</artifactId>
+            <optional>true</optional>
+        </dependency>
     </dependencies>
 
     <build>

--- a/vaadin-spring-tests/test-spring-common/src/main/java/com/vaadin/flow/spring/test/EmbeddedJettyConfiguration.java
+++ b/vaadin-spring-tests/test-spring-common/src/main/java/com/vaadin/flow/spring/test/EmbeddedJettyConfiguration.java
@@ -1,0 +1,25 @@
+package com.vaadin.flow.spring.test;
+
+import javax.servlet.Servlet;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.util.Loader;
+import org.eclipse.jetty.webapp.WebAppContext;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.web.embedded.jetty.JettyServerCustomizer;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+// This class, as well as the optional spring-boot-starter-jetty dependency,
+// can be removed once https://github.com/eclipse/jetty.project/issues/6400
+// is fixed and spring-boot is updated to the latest jetty version
+// containing the fix or spring-boot changed their configuration.
+@Configuration
+@ConditionalOnClass({Servlet.class, Server.class, Loader.class,
+                     WebAppContext.class})
+public class EmbeddedJettyConfiguration {
+
+    @Bean
+    JettyServerCustomizer stopTimeoutCustomizer() {
+        return (server) -> server.setStopTimeout(1);
+    }
+}

